distributed/active_memory_manager.py

```diff
@@ -93,15 +90,25 @@ def __init__(
     def start(self, comm=None) -> None:
         """Start executing every ``self.interval`` seconds until scheduler shutdown"""
         pc = PeriodicCallback(self.run_once, self.interval * 1000.0)
-        self.scheduler.periodic_callbacks["amm"] = pc
+        self.scheduler.periodic_callbacks[f"amm-{id(self)}"] = pc
         pc.start()
```

Comment:
That's likely a fringe edge case but in case this method is called twice, this would also spawn a second PC which would never be cleaned up. Maybe we should check if `amm-{id(self)}` is in `scheduler.periodic_callbacks` already?

Reply:
good catch